multidict/_multilib/parser.h

```diff
@@ -0,0 +1,147 @@
+#ifndef _MULTIDICT_PARSER_H
+#define _MULTIDICT_PARSER_H
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+static int raise_unexpected_kwarg(const char *fname, PyObject* argname)
+{
+    PyErr_Format(PyExc_TypeError,
+                 "%.150s() got an unexpected keyword argument '%.150U'",
+                 fname, argname);
+    return -1;
+}
+
+static int raise_missing_posarg(const char *fname, char* argname)
+{
+    PyErr_Format(PyExc_TypeError,
+                 "%.150s() missing 1 required positional argument: '%.150s'",
+                 fname, argname);
+    return -1;
+}
+
+
+
+/* Parse FASTCALL|METH_KEYWORDS arguments as two args,
+the first arg is mandatory and the second one is optional.
+If the second arg is not passed it remains NULL pointer.
+
+The parser accepts three forms:
+1. all posiotional args,
+2. fist positional, second keyword-arg
+3. all named keyword args.
+*/
+
+static int parse2(const char* fname,
+                  PyObject*const *args,
+                  Py_ssize_t nargs,
+                  PyObject *kwnames,
+                  Py_ssize_t minargs,
+                  const char* arg1name,
+                  PyObject **arg1,
+                  const char* arg2name,
+                  PyObject **arg2
+)
+{
+    assert(maxargs>=1);
+    assert(maxargs<=2);
```

Comment:
`maxargs` ?

Reply:
Or is it better to keep it as-is, these assertions exist for checking that the function was called with a correct compile-time constant.